src/test/java/Homework20.java

```diff
@@ -0,0 +1,78 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class Homework20 extends BaseTest{
+
+    @Test
+    public void deleteEmptyPlaylist() {
+        //navigatetoPage();
+        provideEmail("daria.pavlyuk@testpro.io");
+        providePassword("te$t$tudent");
+        clickSubmit();
+        String playlistName = "Daria's playlist";
+        choosePlaylist(playlistName);
+        deletePlaylist();
+        getDeletePlaylistMessage();
+        assertDeletedPlaylistMessageIsDisplayed();
+
+    }
+    //step1
+    public void navigatetoPage() {
+        String url = baseUrl;
+        driver.get(baseUrl);
+    }
+
+    //step2
+    public void provideEmail (String email){
+        WebElement emailField = driver.findElement(By.cssSelector("[type ='email']"));
+        emailField.click();
+        emailField.sendKeys(email);
+        wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("[type ='email']")));
+    }
+    //step3
+    public void providePassword(String password) {
+        WebElement passwordField = driver.findElement(By.cssSelector("[type='password']"));
+        passwordField.click();
+        passwordField.sendKeys(password);
+        wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("[type='password']")));
+    }
+    //step4
+    public void clickSubmit() {
+        WebElement submit = driver.findElement(By.cssSelector("[type ='submit']"));
+        submit.click();
+        wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("[type ='submit']") ));
```

Comment:
The issue with this code is that after clicking the "submit" element, it immediately waits for the presence of the same "submit" element again using wait.until(). This may lead to a race condition where the element may not have been removed from the DOM or may not yet have the expected properties set by the time wait.until() is executed.

Ensure that you wait for the correct element and then click on it or perform any actions. You don't have to click on the submit button and then again wait for the same button.

The best way is to wait for element and then click on it.
 ` wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("[type ='submit']") ));`
`submit.click();`

Reply:
Please follow the same process in other methods too.